src/main/java/org/snomed/aag/data/domain/CriteriaItemSignOffFactory.java

```diff
@@ -0,0 +1,32 @@
+package org.snomed.aag.data.domain;
+
+import org.springframework.stereotype.Component;
+
+@Component
+public class CriteriaItemSignOffFactory {
+	/**
+	 * Return CriteriaItemSignOff. If branch is the same as ProjectAcceptanceCriteria.branchPath, CriteriaItemSignOffProject will be returned. Otherwise, CriteriaItemSignOffTask
```

Comment:
To make it obvious there is a difference in CriteriaItemSignOff for projects and tasks, I created specific classes for them. I have also added a factory to build each one depending on the branch in question is for the project or task.